app/mainwindow.cpp

```diff
@@ -129,3 +129,9 @@ void MainWindow::on_btnExportFile_clicked()
                                                     "Text files (*.csv)");
     m_main->saveCurrentTableAsCSV(filePath);
 }
+
+void MainWindow::on_btnCopyAll_clicked()
+{
+    LOG_ENTRY;
+    m_main->copyAllTableData();
```

Comment:
If any exception occur, it shouldn't cross slot boundary. Avoid possible program failures by using exception handler here.